mathjax3-ts/output/chtml/Wrapper.ts

```diff
@@ -0,0 +1,604 @@
+        }
+        this.chtml = parent.appendChild(this.html('mjx-' + this.node.kind));
+        return this.chtml;
+    }
+
+    /*
+     * Set the CSS styles for the chtml element
+     */
+    protected handleStyles() {
+        if (this.styles) {
+            const styles = this.styles.cssText;
+            if (styles) {
+                this.chtml.style.cssText = styles;
+            }
+        }
+    }
+
+    /*
+     * Set the CSS for the math variant
+     */
+    protected handleVariant() {
+        if (this.node.isToken && this.variant !== '-explicitFont') {
+            this.chtml.className = VARIANT[this.variant] || VARIANT.normal;
+        }
+    }
+
+    /*
+     * Set the (relative) scaling factor for the node
+     */
+    protected handleScale() {
+        const scale = (Math.abs(this.bbox.rscale - 1) < .001 ? 1 : this.bbox.rscale);
+        if (this.chtml && scale !== 1) {
+            this.chtml.style.fontSize = this.percent(scale);
+        }
+    }
+
+    /*
+     * Add the proper spacing according to the TeX rules
+     *   FIXME:  still need to handle MathML spacing
+     */
+    protected handleSpace() {
+        const space = this.node.texSpacing();
+        if (space) {
+            this.bbox.L = this.length2em(space);
+            this.chtml.setAttribute('space', SPACE[space]);
+        }
+    }
+
+    /*
+     * Add the foreground and background colors
+     * (Only look at explicit attributes, since inherited ones will
+     *  be applied to a parent element, and we will inherit from that)
+     */
+    protected handleColor() {
+        const attributes = this.node.attributes;
+        const mathcolor = attributes.getExplicit('mathcolor') as string;
+        const color = attributes.getExplicit('color') as string;
+        const mathbackground = attributes.getExplicit('mathbackground') as string;
+        const background = attributes.getExplicit('background') as string;
+        if (mathcolor || color) {
+            this.chtml.style.color = mathcolor || color;
+        }
+        if (mathbackground || background) {
+            this.chtml.style.backgroundColor = mathbackground || background;
+        }
+    }
+
+    /*
+     * Copy RDFa, aria, and other tags from the MathML to the CHTML output nodes.
+     * Don't copy those in the skipAttributes list, or anything that already exists
+     * as a property of the node (e.g., no "onlick", etc.).  If a name in the
+     * skipAttributes object is set to false, then the attribute WILL be copied.
+     */
+    protected handleAttributes() {
+        const attributes = this.node.attributes;
+        const globals = attributes.getAllGlobals();
+        const skip = CHTMLWrapper.skipAttributes;
+        for (const name of attributes.getExplicitNames()) {
+            if (skip[name] === false || (!globals.hasOwnProperty(name) && !skip[name] &&
+                                         typeof((this.chtml as {[name: string]: any})[name]) === 'undefined')) {
```

Comment:
I am not sure I fully understand all these cases. Can you spread it out a bit and have some inline comments?

Reply:
This is changed in a later PR, so let's put off these edits until then, in order to avoid merge conflicts in the future.  I'll add some comments then.

Reply:
OK.